File: backend/manager-domain/src/main/kotlin/br/com/doula/manager/infrastructure/PregnantUtils.kt
package br.com.doula.manager.infrastructure

import java.time.LocalDate
import java.time.temporal.ChronoUnit

object PregnantUtils {
    fun calculatePregnancyWeek(lmpDate: LocalDate?, targetDate: LocalDate): Int {
        val gestationalDays = ChronoUnit.DAYS.between(lmpDate, targetDate).toInt()

        return gestationalDays / 7
    }

    fun calculateBirthForecast(lmpDate: LocalDate): LocalDate = lmpDate.plusDays(280)
}

File: backend/utils/pregnant_utils.py
from datetime import date, timedelta

class PregnantUtils:

    @staticmethod
    def calculate_pregnancy_week(lmp_date: date | None, target_date: date) -> int:
        if lmp_date is None:
            return 0
        gestational_days = (target_date - lmp_date).days
        return gestational_days // 7

    @staticmethod
    def calculate_birth_forecast(lmp_date: date) -> date:
        return lmp_date + timedelta(days=280)